src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -637,11 +800,15 @@ impl AccessChecker for PrivilegeAccess {
             }
             Plan::Replace(plan) => {
                 //plan.delete_when is Expr no need to check privileges.
+                let (db_id, table_id) = match self.convert_to_id(&tenant, &plan.catalog, &plan.database, Some(&plan.table)).await? {
+                    ObjectId::Table(db_id, table_id) => { (db_id, Some(table_id)) }
+                    ObjectId::Database(db_id) => { (db_id, None) }
+                };
                 self.validate_access(
```

Comment:
how about adding a function like `validate_access_table`, and put `convert_to_id` inside it, it may help reduce some duplicated `convert_to_id` calls.

Reply:
Yes, we can optimize it.

Reply:
Now split two function : 

validate_table_access,

validate_db_access